sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1627,3 +1627,149 @@ def sdc_pandas_dataframe_groupby_impl(self, by=None, axis=0, level=None, as_inde
+    """
+    Intel Scalable Dataframe Compiler User Guide
+    ********************************************
+    Pandas API: pandas.DataFrame.reset_index
+
+    Limitations
+    -----------
+    Reset the index of the DataFrame, and use the default one instead.
+    Parameters level, inplacem col_level, col_fill unsupported.
+
+    Examples
+    --------
+    .. literalinclude:: ../../../examples/dataframe/dataframe_reset_index_drop_False.py
+        :language: python
+        :lines: 29-
+        :caption: Reset the index of the DataFrame, and use the default one instead.
+                  The old index becomes the first column.
+        :name: ex_dataframe_reset_index
+
+    .. command-output:: python ./dataframe/dataframe_reset_index_drop_False.py
+        :cwd: ../../../examples
+
+    .. literalinclude:: ../../../examples/dataframe/dataframe_reset_index_drop_True.py
+        :language: python
+        :lines: 29-
+        :caption: Reset the index of the DataFrame, and use the default one instead.
+        :name: ex_dataframe_reset_index
+
+    .. command-output:: python ./dataframe/dataframe_reset_index_drop_True.py
+        :cwd: ../../../examples
+
+
+    Intel Scalable Dataframe Compiler Developer Guide
+    *************************************************
+    Pandas DataFrame method :meth:`pandas.DataFrame.reset_index` implementation.
+
+   .. only:: developer
+
+       Test: python -m sdc.runtests -k sdc.tests.test_dataframe.TestDataFrame.test_df_reset_index*
+
+   Parameters
+   -----------
+   self: :class:`pandas.DataFrame`
+       input arg
+   level: :obj:`int`, `str`, `tuple`, or `list`, default None
+       *unsupported*
+   drop: :obj:`bool`, default False
+       Just reset the index, without inserting it as a column in the new DataFrame.
+   inplace: :obj:`bool`, default False
+       *unsupported*
+   col_level: :obj:`int`, `str`, default 0
+       *unsupported*
+   col_fill: :obj, default ''
+       *unsupported*
+
+   Returns
+   -------
+   :obj:`pandas.DataFrame`
+        DataFrame with the new index or None if inplace=True.
+   """
+
+    func_name = 'reset_index'
+
+    ty_checker = TypeChecker('Method {}().'.format(func_name))
+    ty_checker.check(self, DataFrameType)
+
+    if not (isinstance(level, (types.Omitted, types.NoneType)) or level is None):
+        ty_checker.raise_exc(level, 'None', 'level')
+
+    if not isinstance(drop, (types.Omitted, types.Boolean)):
+        ty_checker.raise_exc(drop, 'bool', 'drop')
+
+    if not (isinstance(inplace, (types.Omitted, types.Boolean)) or inplace is False):
+        ty_checker.raise_exc(inplace, 'False', 'inplace')
+
+    if not (isinstance(col_level, (types.Omitted, types.Integer)) or col_level == 0):
+        ty_checker.raise_exc(col_level, '0', 'col_level')
+
+    if not (isinstance(col_fill, (types.Omitted, types.StringLiteral)) or col_fill == ""):
+        ty_checker.raise_exc(col_fill, '""', 'col_fill')
```

Comment:
There's no point in checking types of unsupported params, instead you should check that these parameters were not provided and raise exception if they were, e.g.:
```
    if not (level is None or isinstance(level, types.Omitted)):
        raise TypingError('{} Unsupported parameters. Given inplace: {}'.format(_func_name, level))
```